dali/test/python/reader/test_caffe.py

```diff
@@ -99,6 +99,50 @@ def test_caffe_reader_alias():
     compare_pipelines(new_pipe, legacy_pipe, batch_size_alias_test, 50)
 
 
+def test_caffe_sharding():
+    @pipeline_def(batch_size=1, device_id=0, seed=123, num_threads=1)
+    def pipeline(shard_id, num_shards, stick_to_shard):
+        images, _ = fn.readers.caffe(
+            name='Reader',
+            path=caffe_db_folder,
+            pad_last_batch=True,
+            random_shuffle=False,
+            shard_id=shard_id,
+            stick_to_shard=stick_to_shard,
+            num_shards=num_shards)
+        return images
+
+    def get_data(shard_id, num_shards, stick_to_shard):
+        p = pipeline(shard_id, num_shards, stick_to_shard)
+        p.build()
+        size = p.reader_meta()['Reader']['epoch_size_padded']
+
+        # This should return some unique number for each sample
+        def sample_id(sample):
+            return sample.as_array().sum()
+
+        return {sample_id(p.run()[0]) for _ in range(size * num_shards)}
```

Comment:
```suggestion
        return {sample_id(p.run()[0]) for _ in range(size / num_shards)}
```

Reply:
Fixed in https://github.com/NVIDIA/DALI/pull/5172/commits/851755778a22d97fee12d985de9eba501f255eaa

Reply:
👍